sdk/log/DESIGN.md

```diff
@@ -0,0 +1,183 @@
+# Logs SDK
+
+## Abstract
+
+`go.opentelemetry.io/otel/sdk/log` provides Logs SDK compliant with the
+[specification](https://opentelemetry.io/docs/specs/otel/logs/sdk/).
+
+The main and recommended use case is to configure the SDK to use an OTLP
+exporter with a batch processor.[^1] Therefore, the design aims to be
+high-performant in this scenario.
+
+The prototype was created in
+[#4955](https://github.com/open-telemetry/opentelemetry-go/pull/4955).
+
+## Modules structure
+
+The SDK is published as a single `go.opentelemetry.io/otel/sdk/log` Go module.
+
+The exporters are going to be published as following Go modules:
+
+- `go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploggrpc`
+- `go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploghttp`
+- `go.opentelemetry.io/otel/exporters/stdout/stdoutlog`
+
+## LoggerProvider
+
+The [LoggerProvider](https://opentelemetry.io/docs/specs/otel/logs/sdk/#loggerprovider)
+is defined as follows:
+
+```go
+type LoggerProvider struct {
+	embedded.LoggerProvider
+}
+
+func NewLoggerProvider(...Option) *LoggerProvider
+
+func (*LoggerProvider) Logger(name string, options ...log.LoggerOption) log.Logger
+
+type Option interface { /* ... */ }
+
+func WithResource(*resource.Resource) Option
+```
+
+## LogRecord limits
+
+The [LogRecord limits](https://opentelemetry.io/docs/specs/otel/logs/sdk/#logrecord-limits)
+can be configured using following options:
+
+```go
+func WithAttributeCountLimit(limit int) Option
+
+func WithAttributeValueLengthLimit(limit int) Option
+```
+
+The limits can be also configured using the `OTEL_LOGRECORD_*` environment variables as
+[defined by the specification](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#logrecord-limits).
+
+The options take precedence over environmental variables.
+
+### LogRecordProcessor and LogRecordExporter  
+
+Both [LogRecordProcessor](https://opentelemetry.io/docs/specs/otel/logs/sdk/#logrecordprocessor)
+and [LogRecordExporter](https://opentelemetry.io/docs/specs/otel/logs/sdk/#logrecordexporter)
+are defined via an `Exporter` interface:[^2]
+
+```go
+func WithExporter(Exporter) Option {
+	return nil
+}
+
+type Exporter interface {
+	Export(ctx context.Context, records []*Record) error
+	Shutdown(ctx context.Context) error
+	ForceFlush(ctx context.Context) error
+}
+```
+
+The `Record` struct represents the [ReadWriteLogRecord](https://opentelemetry.io/docs/specs/otel/logs/sdk/#readwritelogrecord).
+
+```go
+type Record struct { /* ... */ }
```

Comment:
Most of these look like they are duplicated from the API's record.  Did you consider embedding the API `Record` here?

Reply:
Attributes manipulation is different.

Reply:
Moreover, I think that `SetAttributes` and `SetBody` should make sure that the keys (in the map) are unique (duplicate keys are not allowed). Per https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#type-mapstring-any

Reply:
Addressed 42eedc10ab7c8d38647b6317ad84b769209328cb